com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest fails occasionally

@htynkn mentions in mailing list:

For now, I can only reproduce this issue in travis ci with jdk9 (refer to logs: https://api.travis-ci.org/v3/job/365133362/log.txt). I also did some test with my own machine:

Mac 10.13.2 with Jdk7, Jdk8 and Jdk9 can't reproduce this issue
Window 8 with jdk8 can reproduce this issue, I run test 1000 times, 247 of them are fail.

The only thing I can think about is hessian.overload.method config for hessian protocol. After I set this value to true in test. I can't reproduce this issue both in travis ci(refer to https://api.travis-ci.org/v3/job/365152345/log.txt) and my own machine.

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
Running com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest
log4j:WARN No appenders could be found for logger (com.alibaba.dubbo.common.logger.LoggerFactory).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by javassist.ClassPool (file:/home/travis/.m2/repository/org/javassist/javassist/3.20.0-GA/javassist-3.20.0-GA.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of javassist.ClassPool
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2018-04-11 14:04:17.687:INFO::Logging to STDERR via org.mortbay.log.StdErrLog
2018-04-11 14:04:17.860:INFO::jetty-6.1.x
2018-04-11 14:04:18.121:INFO::Started SelectChannelConnector@0.0.0.0:5342
2018-04-11 14:04:19.094:WARN::/com.alibaba.dubbo.rpc.protocol.hessian.HessianService: com.caucho.hessian.io.HessianProtocolException: sayHello: expected int at 0x7a (z)
2018-04-11 14:04:19.189:WARN::/com.alibaba.dubbo.rpc.protocol.hessian.HessianService: com.caucho.hessian.io.HessianProtocolException: sayHello: expected int at 0x7a (z)
Tests run: 5, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 2.609 sec <<< FAILURE! - in com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest
testHttpClient(com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest)  Time elapsed: 0.662 sec  <<< ERROR!
com.caucho.hessian.client.HessianRuntimeException: com.caucho.hessian.io.HessianProtocolException: 'ï¿¿' is an unknown code
	at com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest.testHttpClient(HessianProtocolTest.java:83)
Caused by: com.caucho.hessian.io.HessianProtocolException: 'ï¿¿' is an unknown code
	at com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest.testHttpClient(HessianProtocolTest.java:83)

testHessianProtocol(com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest)  Time elapsed: 0.004 sec  <<< ERROR!
com.caucho.hessian.client.HessianRuntimeException: com.caucho.hessian.io.HessianProtocolException: 'ï¿¿' is an unknown code
	at com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest.testHessianProtocol(HessianProtocolTest.java:48)
Caused by: com.caucho.hessian.io.HessianProtocolException: 'ï¿¿' is an unknown code
	at com.alibaba.dubbo.rpc.protocol.hessian.HessianProtocolTest.testHessianProtocol(HessianProtocolTest.java:48)


Results :

Tests in error: 
  HessianProtocolTest.testHessianProtocol:48 Â» HessianRuntime com.caucho.hessian...
  HessianProtocolTest.testHttpClient:83 Â» HessianRuntime com.caucho.hessian.io.H...
```

--- a/dubbo-rpc/dubbo-rpc-hessian/src/test/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocolTest.java
+++ b/dubbo-rpc/dubbo-rpc-hessian/src/test/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocolTest.java
@@ -25,7 +25,7 @@ import com.alibaba.dubbo.rpc.ProxyFactory;
 import com.alibaba.dubbo.rpc.RpcException;
 import com.alibaba.dubbo.rpc.protocol.hessian.HessianServiceImpl.MyException;
 
-import junit.framework.Assert;
+import org.junit.Assert;
 import org.junit.Test;
 
 import static org.junit.Assert.fail;
@@ -41,7 +41,7 @@ public class HessianProtocolTest {
         Assert.assertFalse(server.isCalled());
         ProxyFactory proxyFactory = ExtensionLoader.getExtensionLoader(ProxyFactory.class).getAdaptiveExtension();
         Protocol protocol = ExtensionLoader.getExtensionLoader(Protocol.class).getAdaptiveExtension();
-        URL url = URL.valueOf("hessian://127.0.0.1:5342/" + HessianService.class.getName() + "?version=1.0.0");
+        URL url = URL.valueOf("hessian://127.0.0.1:5342/" + HessianService.class.getName() + "?version=1.0.0&hessian.overload.method=true");
         Exporter<HessianService> exporter = protocol.export(proxyFactory.getInvoker(server, HessianService.class, url));
         Invoker<HessianService> invoker = protocol.refer(HessianService.class, url);
         HessianService client = proxyFactory.getProxy(invoker);
@@ -76,7 +76,7 @@ public class HessianProtocolTest {
         Assert.assertFalse(server.isCalled());
         ProxyFactory proxyFactory = ExtensionLoader.getExtensionLoader(ProxyFactory.class).getAdaptiveExtension();
         Protocol protocol = ExtensionLoader.getExtensionLoader(Protocol.class).getAdaptiveExtension();
-        URL url = URL.valueOf("hessian://127.0.0.1:5342/" + HessianService.class.getName() + "?version=1.0.0&client=httpclient");
+        URL url = URL.valueOf("hessian://127.0.0.1:5342/" + HessianService.class.getName() + "?version=1.0.0&client=httpclient&hessian.overload.method=true");
         Exporter<HessianService> exporter = protocol.export(proxyFactory.getInvoker(server, HessianService.class, url));
         Invoker<HessianService> invoker = protocol.refer(HessianService.class, url);
         HessianService client = proxyFactory.getProxy(invoker);
@@ -100,7 +100,7 @@ public class HessianProtocolTest {
             client.timeOut(6000);
             fail();
         } catch (RpcException expected) {
-            Assert.assertEquals(true, expected.isTimeout());
+            Assert.assertTrue(expected.isTimeout());
         } finally {
             invoker.destroy();
             exporter.unexport();
@@ -121,6 +121,7 @@ public class HessianProtocolTest {
             client.customException();
             fail();
         } catch (MyException expected) {
+
         }
         invoker.destroy();
         exporter.unexport();

